apis/infrastructure/v1beta1/byocluster_webhook_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubectl/pkg/scheme"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("ByoclusterWebhook", func() {
+	Context("When ByoCluster gets a create request", func() {
+		var (
+			byoCluster        *ByoCluster
+			ctx               context.Context
+			k8sClientUncached client.Client
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+			var clientErr error
+			k8sClientUncached, clientErr = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+			Expect(clientErr).NotTo(HaveOccurred())
+
+			byoCluster = &ByoCluster{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ByoCluster",
+					APIVersion: clusterv1.GroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "byocluster-create",
+					Namespace: "default",
+				},
+				Spec: ByoClusterSpec{},
+			}
+		})
+
+		It("should reject the request when BundleLookupTag is empty", func() {
```

Comment:
I think we can test for 2 cases here.
1. when `BundleLookupTag` is not present
2. when `BundleLookupTag` is empty

It should reject request in both cases.

Reply:
Also another case when there is no error - positive scenario.

Reply:
This second 

> I think we can test for 2 cases here.
> 
>     1. when `BundleLookupTag` is not present
> 
>     2. when `BundleLookupTag` is empty
> 
> 
> It should reject request in both cases.

For Go-programming, actually there is no such "one field not present". All field are all present, it just initialize or not.

Reply:
I see, okay. Sounds good then. Thanks!

Reply:
> Also another case when there is no error - positive scenario.

Ok.